src/vs/editor/contrib/referenceSearch/browser/referencesWidget.ts

```diff
@@ -212,8 +212,19 @@ class DataSource implements tree.IDataSource {
 	}
 }
 
+/**
+ * RevealedReference wraps an optional OneReference.
+ * This allows RevealedReference to be passed around (i.e. to Controller)
+ * and updates to the reference can be seen.
+ */
+type RevealedReference = { ref: OneReference | undefined };
```

Comment:
The Controller needs to able to write the reference (to clear it). The lifecycle of the revealed reference is independent of the Controller.

See lines 280:283
https://sourcegraph.com/github.com/sourcegraph/vscode@fdb48b853513493c53e3dc6beb881d717b6de76e/-/blob/src/vs/editor/contrib/referenceSearch/browser/referencesWidget.ts#L280:1-284:1